### PR TITLE
add freebsd data

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: lib/beaker-hostgenerator/data.rb [hunk 1]
```diff
@@ -685,6 +685,22 @@ def osinfo
             'platform' => 'fedora-14-i386',
           },
         },
+        'freebsd12-64' => {
+          :general => {
+            'platform' => 'freebsd-12-amd64',
```
Comment (evgeni) [inline lib/beaker-hostgenerator/data.rb:690]: I don't exactly understand the "platform" string. `amd64` is FreeBSD way to say x86_64 (like Debian), but in other places I've seen this still being used as `freebsd-123-x86_64`, so no idea…

Comment (bastelfreak) [reply lib/beaker-hostgenerator/data.rb:690]: I always struggle with this myself and need to dig through the code or wait for the awesome ewoud...

Comment (evgeni) [reply lib/beaker-hostgenerator/data.rb:690]: the failures in https://github.com/voxpupuli/puppet-mosquitto/actions/runs/4269959304/jobs/7433482798 used `freebsd-123-amd64`, so I am going to stick with that for now.

Comment (evgeni) [reply lib/beaker-hostgenerator/data.rb:690]: also https://github.com/voxpupuli/rspec-puppet-facts/blob/1418bca958049060a77dd49f10a9a298d15dc438/lib/rspec-puppet-facts.rb#L71-L72

**Review by bastelfreak — APPROVED** [review]

File: lib/beaker-hostgenerator/data.rb [hunk 2]
```diff
@@ -685,6 +685,22 @@ def osinfo
...
+          },
+          :vagrant => {
+            'box' => 'generic/freebsd12',
```
Comment (ekohl) [inline lib/beaker-hostgenerator/data.rb:693]: In https://github.com/voxpupuli/beaker-hostgenerator/blob/master/lib/beaker-hostgenerator/hypervisor/vagrant.rb we already do similar conversions automatically so you don't have to maintain a manual mapping and you get it "for free" when FreeBSD 14 comes out..

**Review by ekohl — COMMENTED** [review]
I'd like to move more to a generated model rather than maintaining such a massive list.

File: lib/beaker-hostgenerator/data.rb (hunk 2 above)
Comment (evgeni) [reply lib/beaker-hostgenerator/data.rb:693]: you're saying I can omit the `vagrant => {}` section?

**Comment by evgeni** [comment]
@ekohl seems to work well, also flipped to actually "generate" the entries too, as they are all identical

File: lib/beaker-hostgenerator/data.rb [hunk 3]
```diff
@@ -147,6 +147,15 @@ def osinfo
         }
       end
 
+      # FreeBSD
+      (11..13).each do |release|
```
Comment (smortex) [inline lib/beaker-hostgenerator/data.rb:151]: FreeBSD 11 is EOL, no need to add it here I think.

**Review by smortex — APPROVED** [review]

**Review by ekohl — APPROVED** [review]
Thanks!

**Pull request merged**